### PR TITLE
Feature/one click new brand

### PR DESCRIPTION
**Pull request by RogerTangos**
draft PR so that I can comment on this

### CODE REVIEW

File: datasource/admin.py [hunk 1]
```diff
@@ -229,6 +229,32 @@ class SwitchitAdmin(DatasourceAdmin, admin.ModelAdmin):
 class UsnicAdmin(DatasourceAdmin, admin.ModelAdmin):
     form = CountriesWidgetOverrideForm
 
+    actions = ['add_to_Brands']
+
+    @admin.action(description='Create new related Brand and copy data')
+    def add_to_Brands(self, request, queryset):
+        self.message_user(request, 'Successfully added new brand', messages.SUCCESS)
```
Comment (RogerTangos) [inline datasource/admin.py:236]: The success message should probably go to the bottom, to make sure it only displays if everything went well?

**Review by RogerTangos — COMMENTED** [review]
This code largely works with the caveates above. 

One other potential issue I see here is that sometimes USNIC have children, as expressed through the JSON blob in the `usnic.control` field. (unfortunately, children point to parents rather than parents to children, which can make this difficult.)

This PR should find all children and warn the user that those children may also need to be related to the brand. Maybe it can do that using a query linked in the success message?

**Comment by RogerTangos** [comment]
Another thing: it would probably be worth doing this one TDD-style to make sure it doesn't break in the future. I imagine that the dev time will be roughly the same and can help with that if you need

**Comment by Kyle0** [comment]
Added checking for child entities (one level down only, does not check for children of children and so on). Involves checking entire Usnic table for each selected, has noticeably slowed entire brand copying process. Horrendous multi-level loop and condition checking invoked to achieve this end without constant erroring. Possibly try...except may be a better, more friendly solution.

File: brand/tests/test.py [hunk 2]
```diff
@@ -40,63 +58,87 @@ def test_create_spelling_dictionary(self):
...
-#         )
+        # Check that no new Brands are created if all Usnic entries selected already exist
+        existing_brands, successful_brands = Brand.create_brand_from_usnic(
+            Usnic.objects.all().values()[:2]
+        )
+        self.assertEqual(len(successful_brands), 0)  # Ensure only one new brand created
+        self.assertEqual(len(existing_brands), 2)  # Check that existing copies of usnic are ignored
+
+
+class BrandDatasourceTestCase(TestCase):
```
[the remark this answers is not in the document]
Comment (Kyle0) [reply brand/tests/test.py:105]: This was there previously and falls out the scope of the current issue. Another ticket is required to make sure it doesn't break anything else if it is removed.

**Pull request closed without merging**